lib/editor/components/timetable/TimetableEditor.js

```diff
@@ -220,11 +220,22 @@ export default class TimetableEditor extends Component<Props, State> {
           cumulativeTravelTime += +halt.flexDefaultTravelTime
 
           // $FlowFixMe Flow doesn't recognize our check for stopId property?
-          objectPath.set(newRow, `stopTimes.${i}.stopId`, patternHaltIsLocation(halt) ? halt.locationId : halt.locationGroupId)
+          const haltId = patternHaltIsLocation(halt) ? halt.locationId : halt.locationGroupId
+          const nextHalt = i + 1 !== patternHalts.length ? patternHalts[i + 1] : null
```

Comment:
Shouldn't the check be i + 2? Otherwise won't the last halt not be included

Reply:
we're checking the nextHalt, so we want to stop when we hit the current halt at the end of the array, so i = length - 1, because that's where there's no "nextHalt"